docs/guides/advanced/deepkit-type.md

```diff
@@ -0,0 +1,96 @@
+---
+id: using-with-deepkit-type
+title: 'Using with @deepkit/type'
+---
+
+This guide shows how to use `typegoose` with `@deepkit/type`.
+
+## Implementation
+
+Suppose you have this `Account` class decorated with `@deepkit/type`:
+
+```ts
+import { classToPlain, t, plainToClass } from '@deepkit/type';
+import { getModelForClass, prop } from '../../src/typegoose';
+
+enum Group {
+  confidential = 'confidential',
+  public = 'public',
+}
+
+class Account {
+  @t.mongoId.group(Group.public)
+  public _id: string;
+
+  @t.group(Group.public)
+  public __v: number;
+
+  @t.group(Group.public)
+  @prop()
+  public email: string;
+
+  @t.group(Group.confidential)
+  @prop()
+  public confidentialProperty?: string;
+}
+
+const AccountModel = getModelForClass(Account);
+```
+
+You can then use, for example:
+
+* `lean()` on the query:
+
+  ```ts
+  const pojo = await AccountModel.findById(id).orFail().lean().exec();
+  // groupsExclude option excludes the property group
+  const access = { groupsExclude: [Group.confidential] };
+  // deserialize pojo back to an Account instance
+  const deserialized = plainToClass(Account, pojo, access);
+ 
+  expect(deserialized).toStrictEqual(accountClassObject);
+  ```
+
+* or a normal document:
+
+  ```ts
+  // exec returns a Mongoose Object
+  const doc = await AccountModel.findById(id).orFail().exec();
+  const access = { groupsExclude: [Group.confidential] };
+  // serialize Account instance back to a Plain Old Javascript Object
+  const serialized = classToPlain(Account, doc, access);
```

Comment:
is it right to use `classToPlain` in this example (where in for the lean example `plainToClass` is used)?

this does not seem to result in the same as the lean example?

(lean example: POJO -> Class | normal example: Document -> POJO (and not Class))

Reply:
Hmm... yeah. Like I said in Discord, the tests don't really show the proper intention of when to use serialization and deserialization of these libraries. In effect, we should be showing an example of a DTO coming in being transformed to a Class to be stored via Typegoose and a Doc coming from Typegoose being transformed to a POJO/ DTO. 

I can rework the docs.

Reply:
resolved with 1fbe58064faecc415eb50c9f2ea0a27afa5bdc19